contracts/gateway/GatewayBase.sol

```diff
@@ -257,6 +262,8 @@ contract GatewayBase is Organized {
         );
 
         storageRoots[_blockHeight] = storageRoot;
+        uint256 oldestStoredBlockHeight = CircularBufferUint.store(_blockHeight);
+        delete storageRoots[oldestStoredBlockHeight];
```

Comment:
is this required in the constructor? Before this, no storage has been initialized right?

Reply:
This will work even if the storageRoot is not available for the given block height.
What do you propose here? What should be initialized?

Reply:
I mean this delete statement is extra here, it's not needed. right?

Reply:
This is not in the constructor, this is in `proveGateway`. Here, it is needed to restrict the storage usage of the contract.

Reply:
Oh yes, My bad. I misunderstood.